mmengine/utils/progressbar_rich.py

```diff
@@ -0,0 +1,214 @@
+        >>>     bar.update()
+        >>>     time.sleep(1)
+    """
+
+    def __init__(self):
+        self.bar = Progress()
+        self.tasks = []
+        self.descriptions = []
+        self.colors = []
+        self.timer = None
+        self.completed = 0
+        self.infinite = False
+
+    @staticmethod
+    def write(msg: str, color: str = 'blue'):
+        """Write the massage."""
+        rich.print(f'[{color}]{msg}')
+
+    def add_task(self,
+                 total: int = None,
+                 color: str = 'blue',
+                 description: str = 'Process...') -> int:
+        """Adding tasks to RichProgressbar.
+
+        Args:
+            total (int): Number of total steps, When the input is None,
+             it indicates a task with unknown length.
+            color (str): Color of progress bar.
+            description (str): Description of progress bar.
+        """
+        if total is not None:
+            assert not self.infinite, (
+                'The prior task is an infinite task (total is None), '
+                'RichProgressBar can only accept one infinite task')
+
+            if total <= 0:
+                raise ValueError(
+                    'Total only exists if it is greater than zero or None.')
+            self.tasks.append(
+                self.bar.add_task(
+                    f'[{color}]{description}_0/{total}', total=total))
+            self.colors.append(color)
+            self.descriptions.append(description)
+            task_id = self.bar.tasks[-1].id
+        else:
+            assert not self.tasks, (
+                'Since the total argument is None, this task is considered '
+                'infinite. RichProgressBar should not have any other tasks '
+                'added to it when an infinite task is present.')
+            self.write('completed: 0, elapsed: 0s', color)
+            self.timer = Timer()
+            self.infinite = True
+            self.completed = 0
+            self.tasks.append(0)
+            self.colors.append(color)
+            self.descriptions.append(description)
+            task_id = 0
+        return task_id
+
+    def update(self, task_id: int = 0, advance: int = 1):
+        """update progressbar.
+
+        Args:
+            task_id (int): Task ID that needs to be updated.
+            advance (int): Update step size.
+        """
+        if advance <= 0:
+            raise ValueError('advance should greater than zero.')
+
+        if self.infinite:
+            self.completed += advance
+            elapsed = self.timer.since_start()
+            if elapsed > 0:
+                fps = self.completed / elapsed
+            else:
+                fps = float('inf')
+            self.write(
+                f'completed: {self.completed}, '
+                f'elapsed: {int(elapsed + 0.5)}s, '
+                f'{fps:.1f} tasks/s', self.colors[task_id])
+        else:
+            assert task_id >= -1 and task_id < len(self.tasks), (
```

Comment:
Accepting a non-negative task_id is sufficient. If you want `task_id` to be a negative value, its value should not be restricted to -1, and its meaning should be consistent with the index of the `list`.